pilz_trajectory_generation/src/command_list_manager.cpp

```diff
@@ -108,7 +108,8 @@ bool CommandListManager::checkRadiiForOverlap(const robot_trajectory::RobotTraje
     return false;
   }
 
-  const std::string& blend_frame {getTipFrame(model_->getJointModelGroup(traj_A.getGroupName()))};
+  assert(!model_->getJointModelGroup(traj_A.getGroupName())->isEndEffector());
```

Comment:
Thanks for your follow-up.

Is this assertion necessary? You already check for the existence of a solver in https://github.com/PilzDE/pilz_industrial_motion/blob/3f6150551143d83f9ea30fe883f99ce590375c4e/pilz_trajectory_generation/include/pilz_trajectory_generation/tip_frame_getter.h#L25. Why is an end-effector not allowed here? The PG+70 does not have a solver defined an is thus excluded.

Reply:
You are right, It is not necessary.
Imho it would be nice, if the user obtains an error message like e.g. "blending of an end-effector movement is not supported."

Reply:
Indeed a Warning / Error message is more user-friendly than an assertion.

But on the other hand: What would be necessary, to allow end-effector-blending for groups that have a solver? Is there anything missing or it this just a defensive "we've not tested it"?

Reply:
I think we have a huge misunderstanding when it comes to "assertions". So let's just talk about it for a second:
Assertions are things that should never ever happen. However, I might be in a function where I assume that a certain assumptions holds, which, for example, is checked beforehand. To ensure that the programmer reading the functions knows about this implicit assumption I add an assertion. Also, I want to ensure that the conditions always holds, even if somebody accidentally deletes the check which should be performed beforehand.

Now, back to the concrete problem:
Yes, I actually agree with both of you and more importantly the check and the warning already exist [here](https://github.com/PilzDE/pilz_industrial_motion/blob/bd5a71ccb64559fbd050525b761af6b7afa70fb0/pilz_trajectory_generation/src/command_list_manager.cpp#L181). However, like stated before, this is not the reason for the assertion. The assertion exists for the next programmer reading this lines and for the absolute worst case scenario that somebody accidentally deletes the end-effector check beforehand.

@jschleicher 
A pure end-effector blending does not make much sense for me. Why should I want to blend a gripper open or close movement?
The only exception I can think of is the case where you have a group consisting of a manipulator + end-effector. But in this case should work because, in that case, the group is not a pure end-effector.

Reply:
Thanks @hslusarek. I missed the check you mention.

Reply:
It seems one can have different opinion whether a blending of pure end-effector groups makes sense or not. Regardless, why should we categorically exclude it, despite our code is capable of it?

I suggest that we do not care, if a group is an end-effector [here](https://github.com/PilzDE/pilz_industrial_motion/blob/bd5a71ccb64559fbd050525b761af6b7afa70fb0/pilz_trajectory_generation/src/command_list_manager.cpp#L181), but only require that a solver exists. Why not? Then the asserts can be omitted definitely.

Reply:
End-effectors can be multi-joint groups (like articulated hands) and there blending is as useful as it is for the serial manipulator.

Reply:
Ok. I replaced the assumption that end-effector cannot be blended with the more general assumptions that only groups without solver cannot be blended. However, solver allow more than one tip frame. What should we do in this case? Simply take the first one? Return with error?